modules/packages/LinearAlgebra.chpl

```diff
@@ -893,32 +893,33 @@ proc diag(A: [?Adom] ?eltType, k=0) {
 
 private proc _diag_vec(A:[?Adom] ?eltType) {
   const (m, n) = Adom.shape;
-  const d = if m < n then 0 else 1;
-  const dim = Adom.dim(d);
-
-  var diagonal = Vector(dim, eltType);
+  const diagSize = if m < n then m else n;
 
-  forall i in dim do
-    diagonal[i] = A[i,i];
+  var diagonal : [0..#diagSize] eltType;
```

Comment:
I think the new vector should inherit the domain from `A`. I would assume we inherit from the smaller dimension of `A`.